frontend/src/components/common/SimpleTable.tsx

```diff
@@ -125,6 +125,12 @@ export default function SimpleTable(props: SimpleTableProps) {
     const tableNameUrlParam = getFilterValueByNameFromURL('tableName', history) || null;
     if (!tableNameUrlParam || tableNameUrlParam === tableName) {
       let rowsPerPageVal: number = parseInt(getFilterValueByNameFromURL('rowsPerPage', location) as string) || rowsPerPageOptions[0];
+      if (data && rowsPerPageVal && data.length < rowsPerPageVal) {
+        // get me the nearest rowsPerPageVal from the rowsPerPageOptions
+        rowsPerPageVal = rowsPerPageOptions.sort((a, b) => {
+          return Math.abs(a - rowsPerPageVal) - Math.abs(b - rowsPerPageVal);
+        })[0];
+      }
```

Comment:
This check is missing the page value, as if we use `rowsPerPage=2&page=30000` with a data length of 3, it's the page that will create problems.
So the check should be`if (data && data.length < (rowsPerPageVal * pageVal))`, and the action in case this is true should just be:
1) Use the default `rowsPerPage` if the value provided is not in `rowsPerPageOptions`;
2) And show the last page of the table (i.e. floor of `data.length / rowsPerPage`).